src/sqlfluff/dialects/dialect_hive.py

```diff
@@ -979,24 +975,6 @@ class SetExpressionSegment(ansi.SetExpressionSegment):
     )
 
 
-class OrderByClauseSegment(ansi.OrderByClauseSegment):
-    """A `ORDER BY` clause like in `SELECT`."""
-
-    match_grammar = ansi.OrderByClauseSegment.match_grammar.copy()
-    match_grammar.terminator = OneOf(  # type: ignore
-        "CLUSTER",
-        "DISTRIBUTE",
-        "SORT",
-        "LIMIT",
-        "HAVING",
-        "QUALIFY",
-        # For window functions
-        "WINDOW",
-        Ref("FrameClauseUnitGrammar"),
-        "SEPARATOR",
-    )
-
-
 class ClusterByClauseSegment(ansi.OrderByClauseSegment):
```

Comment:
For reviewers: The ANSI dialect changed the grammar it used a while ago, and it's not one that allows terminators, so this code was redundant.